packages/common/src/createContract.ts

```diff
@@ -150,13 +158,43 @@ export function createContract<
               }
             >getFunctionParameters(parameters as any);
 
-            const request = {
+            const functionSignature = getAbiItem<Abi, string>({
+              abi,
+              name: functionName,
+            }) as AbiFunction;
+            const functionSelector = getFunctionSelector(functionSignature);
+            if (!functionSignature) throw new Error(`Unable to get function signature for ${functionName}`);
+
+            let request: WriteContractParameters = {
               address,
               abi,
               functionName,
               args,
               ...options,
             };
+            const resourceSelector = await getResourceSelector(functionSelector);
+            const shouldUseCallFrom = resourceSelector && resourceSelector !== staticAbiTypeToDefaultValue.bytes32;
+
+            // if the function is not part of the world contract and needs to be routed
+            // to other systems, route it through callFrom
+            if (shouldUseCallFrom) {
+              // TODO figure out how to strongly type Abi
+              const funcSelectorAndArgs = encodeFunctionData<Abi, string>({
+                abi,
+                functionName: functionName,
+                args,
+              });
+              // delegator, resourceSelector, funcSelectorAndArgs
+              const argsForCallFrom = [walletClient.account.address, resourceSelector, funcSelectorAndArgs]; // TODO replace address with delegator
+
+              request = {
+                address,
+                abi,
+                functionName: "callFrom",
+                args: argsForCallFrom,
+                ...options,
+              };
+            }
 
             const result = write(request);
```

Comment:
since we made this function `async`, I'd like to add a note here about intentionally not using `await` here?

```suggestion
            // We intentionally don't use `await` here so we can pass the `result` promise immediately to the `onWrite` call and consumers (i.e. dev tools) can use the write request before it hits the RPC.
            const result = write(request);
```